crates/light-client-prover/src/circuit.rs

```diff
@@ -99,35 +104,52 @@ pub fn run_circuit<DaV: DaVerifier, G: ZkvmGuest>(
-                                Err(_) => continue,
-                            };
-
-                        // Do not add if last l2 height is smaller or equal to previous output
-                        // This is to defend against replay attacks, for example if somehow there is the script of batch proof 1 we do not need to go through it again
-                        if batch_proof_output.last_l2_height <= last_l2_height {
+                        if process_complete_proof::<DaV, G>(
+                            proof,
+                            batch_proof_method_id,
+                            last_l2_height,
+                            &mut initial_to_final,
+                        )
+                        .is_err()
+                        {
                             continue;
                         }
-
-                        recursive_match_state_roots(
-                            &mut initial_to_final,
-                            &BatchProofInfo::new(
-                                batch_proof_output.initial_state_root,
-                                batch_proof_output.final_state_root,
-                                batch_proof_output.last_l2_height,
-                            ),
-                        );
                     }
-                    DaDataLightClient::Aggregate(_) => todo!(),
-                    DaDataLightClient::Chunk(_) => todo!(),
+                    DaDataLightClient::Aggregate(_tx_ids, wtx_ids) => {
+                        let existing_tx_ids: BTreeSet<[u8; 32]> =
+                            wtxid_data.keys().cloned().collect();
+                        let aggregate_tx_ids: BTreeSet<[u8; 32]> =
+                            wtx_ids.iter().cloned().collect();
+
+                        // If we have all the chunks, perform verification
+                        if aggregate_tx_ids.is_subset(&existing_tx_ids) {
+                            // Concatenate complete proof
+                            let complete_proof = wtx_ids
+                                .iter()
+                                .filter_map(|k| wtxid_data.get(k).cloned())
+                                .flatten()
+                                .collect::<Vec<_>>();
+
+                            if process_complete_proof::<DaV, G>(
+                                complete_proof,
+                                batch_proof_method_id,
+                                last_l2_height,
+                                &mut initial_to_final,
+                            )
+                            .is_err()
+                            {
+                                continue;
+                            }
+
+                            for tx_id in &aggregate_tx_ids {
+                                wtxid_data.remove(tx_id);
+                            }
+                        }
+                    }
+                    DaDataLightClient::Chunk(chunk) => {
+                        wtxid_data
```

Comment:
realized there is a dos attack vector with our aproach. let me think of a solution